src/mlpack/methods/linear_svm/linear_svm_main.cpp

```diff
@@ -0,0 +1,419 @@
+    "number of allowed iterations, and the " +
+    PRINT_PARAM_STRING("tolerance") + " parameter specifies the tolerance for "
+    "convergence.  For the parallel SGD optimizer, the " +
+    PRINT_PARAM_STRING("step_size") + " parameter controls the step size taken "
+    "at each iteration by the optimizer and the maximum number of epochs "
+    "(specified with " + PRINT_PARAM_STRING("epochs") + "). If the "
+    "objective function for your data is oscillating between Inf and 0, the "
+    "step size is probably too large.  There are more parameters for the "
+    "optimizers, but the C++ interface must be used to access these."
+    "\n\n"
+    "Optionally, the model can be used to predict the labels for another "
+    "matrix of data points, if " + PRINT_PARAM_STRING("test") + " is "
+    "specified.  The " + PRINT_PARAM_STRING("test") + " parameter can be "
+    "specified without the " + PRINT_PARAM_STRING("training") + " parameter, "
+    "so long as an existing linear SVM model is given with the " +
+    PRINT_PARAM_STRING("input_model") + " parameter.  The output predictions "
+    "from the linear SVM model may be saved with the " +
+    PRINT_PARAM_STRING("predictions") + " parameter." +
+    "\n\n"
+    "As an example, to train a LinaerSVM model on the data '" +
+    PRINT_DATASET("data") + "' with labels '" + PRINT_DATASET("labels") + "' "
+    "with L2 regularization of 0.1, saving the model to '" +
+    PRINT_MODEL("lsvm_model") + "', the following command may be used:"
+    "\n\n" +
+    PRINT_CALL("linear_svm", "training", "data", "labels", "labels",
+        "lambda", 0.1, "delta", 1.0, "number_of_classes", 0,
+        "output_model", "lsvm_model") +
+    "\n\n"
+    "Then, to use that model to predict classes for the dataset '" +
+    PRINT_DATASET("test") + "', storing the output predictions in '" +
+    PRINT_DATASET("predictions") + "', the following command may be used: "
+    "\n\n" +
+    PRINT_CALL("linear_svm", "input_model", "lsvm_model", "test", "test",
+        "predictions", "predictions"),
+    SEE_ALSO("@random_forest", "#random_forest"),
+    SEE_ALSO("@logistic_regression", "#logistic_regression"),
+    SEE_ALSO("LinearSVM on Wikipedia",
+        "https://en.wikipedia.org/wiki/Support-vector_machine"),
+    SEE_ALSO("mlpack::svm::LinearSVM C++ class documentation",
+        "@doxygen/classmlpack_1_1svm_1_1LinearSVM.html"));
+
+// Training parameters.
+PARAM_MATRIX_IN("training", "A matrix containing the training set (the matrix "
+    "of predictors, X).", "t");
+PARAM_UROW_IN("labels", "A matrix containing labels (0 or 1) for the points "
+    "in the training set (y).", "l");
+
+// Optimizer parameters.
+PARAM_DOUBLE_IN("lambda", "L2-regularization parameter for training.", "L",
+    0.0001);
+PARAM_DOUBLE_IN("delta", "Margin of difference between correct class and other "
+    "classes.", "d", 1.0);
+PARAM_INT_IN("number_of_classes", "Number of classes for classification; if "
```

Comment:
I think for consistency with the rest of the codebase, we should call this `num_classes`.